src/ArduinoCloudThing.h

```diff
@@ -40,6 +40,31 @@ static long const MINUTES   = 60;
 static long const HOURS     = 3600;
 static long const DAYS      = 86400;
 
+/******************************************************************************
+ * SYNCHRONIZATION CALLBACKS
+ ******************************************************************************/
+
+#define AUTO_SYNC onAutoSync
```

Comment:
Is there any reason for using those defines? Stroustrop's opinion on macros (somewhat simplified by me): Don't use them unless you have to ( http://www.stroustrup.com/bs_faq2.html#macro ). If you really want the function call to look like `AUTO_SYNC` then I'd suggest you just define it that way `void AUTO_SYNC(ArduinoCloudProperty<T> property)`.

Reply:
I'd prefer to keep the function name as is. The defines are there in order to align with the way in which in the auto generated thingProperties.h the UI configurable properties characteristics are defined. See macro [here](https://github.com/ilcato/ArduinoIoTCloud/blob/d532185eaa4c523458361fb0592467d1c4ff29d7/src/ArduinoIoTCloud.h#L91).
A sample property declaration is:
`  ArduinoCloud.addProperty(Switch, READWRITE, ON_CHANGE, onSwitchChange, AUTO_SYNC);
`